App/src/GameBoard.h

```diff
@@ -1,80 +1,126 @@
 #pragma once
 #include <array>
-#include <bitset>
 #include <cstdint>
+#include <functional>
 #include <iostream>
+#include <limits>
+#include <memory>
 #include <unordered_map>
 #include <utility>
 
-class PairHash {
-public:
-  template <class T1, class T2>
-  std::size_t operator()(const std::pair<T1, T2> &p) const {
-    auto h1 = std::hash<T1>{}(p.first);
-    auto h2 = std::hash<T2>{}(p.second);
+struct ChunkKey {
+  ChunkKey(int32_t x, int32_t y) : x(x), y(y) {}
+  ChunkKey(std::pair<int32_t, int32_t> p) : x(p.first), y(p.second) {}
```

Comment:
If the types are homogenous then you might as well use a std::array<int, 2> or whatever the syntax is. Still takes a brace initializer if that's what you want, and it's a bit more performant.